src/OSPSuite.SimModel/src/QuantityReference.cpp

```diff
@@ -134,38 +137,53 @@ void QuantityReference::XMLFinalizeInstance (const XMLNode & pNode, Simulation *
 {
 	const char * ERROR_SOURCE = "QuantityReference::XMLFinalizeInstance";
 
-	if (!IsTime()) // "Time" is not a real quantity
-	{
-		_quantity = sim->AllQuantities().GetObjectById(_quantityId);
+	if (IsTime()) // "Time" is not a real quantity
+		return;
 
-		if (_quantity == NULL)
-			throw ErrorData(ErrorData::ED_ERROR, ERROR_SOURCE, _parentFormulaInfo+": object with id="+XMLHelper::ToString(_quantityId)+" not found");
+	_quantity = sim->AllQuantities().GetObjectById(_quantityId);
 
-		if (_isReference) //quantity comes from the general <ReferenceList>
-		{
-			if (dynamic_cast<Species *>(_quantity) != NULL)
-				_isSpecies = true;
-			else if (dynamic_cast<Parameter *>(_quantity) != NULL)
-				_isParameter = true;
-			else if (dynamic_cast<Observer *>(_quantity) != NULL)
-				_isObserver = true;
-			else
-				throw ErrorData(ErrorData::ED_ERROR, ERROR_SOURCE, _parentFormulaInfo+": object with id="+XMLHelper::ToString(_quantityId)+" has invalid type");
-
-			_isReference = false;
-		}
+	if (_quantity == NULL)
+		throw ErrorData(ErrorData::ED_ERROR, ERROR_SOURCE, _parentFormulaInfo+": object with id="+XMLHelper::ToString(_quantityId)+" not found");
+
+	if (_isReference) //quantity comes from the general <ReferenceList>
+	{
+		if (dynamic_cast<Species *>(_quantity) != NULL)
+			_isSpecies = true;
+		else if (dynamic_cast<Parameter *>(_quantity) != NULL)
+			_isParameter = true;
 		else
 		{
-			if (_isSpecies && (dynamic_cast<Species *>(_quantity) == NULL))
-				throw ErrorData(ErrorData::ED_ERROR, ERROR_SOURCE, _parentFormulaInfo+": object with id="+XMLHelper::ToString(_quantityId)+" should be a species");
-
-			if (_isParameter && (dynamic_cast<Parameter *>(_quantity) == NULL))
-				throw ErrorData(ErrorData::ED_ERROR, ERROR_SOURCE, _parentFormulaInfo+": object with id="+XMLHelper::ToString(_quantityId)+" should be a parameter");
+			//last possible case: observer
+			auto observer = dynamic_cast<Observer *>(_quantity);
+			if (observer == NULL)
+				throw ErrorData(ErrorData::ED_ERROR, ERROR_SOURCE, _parentFormulaInfo + ": object with id=" + XMLHelper::ToString(_quantityId) + " has invalid type");
 
-			if (_isObserver && (dynamic_cast<Observer *>(_quantity) == NULL))
-				throw ErrorData(ErrorData::ED_ERROR, ERROR_SOURCE, _parentFormulaInfo+": object with id="+XMLHelper::ToString(_quantityId)+" should be a observer");
+			observer->SetIsUsedInFormulas(true);
```

Comment:
If current quantity reference is an observer: this observer is used in some formula